Documentation/Logging.md

```diff
@@ -0,0 +1,90 @@
+# Logging
+
+One of the most helpful components of an actionable bug report is a complete set of logs. Aardvark makes it simple to add logs with the logging utilities in the CoreAardvark framework.
+
+## Getting Started
+
+The easiest way to get started logging with Aardvark is by replacing calls to `print` with `log` in Swift and calls to `NSLog` with `ARKLog` in Objective-C.
+
+```swift
+log("Hello world")
+```
+
+```objc
+ARKLog(@"Hello world");
+```
+
+By default, this will log messages to the Aardvark log store _instead of_ the console. If you would like the messages to be logged to the console as well, toggle the default log store's `printsLogsToConsole` property:
+
+```swift
+ARKLogDistributor.default().defaultLogStore.printsLogsToConsole = true
+```
+
+## Adding Parameters
+
+A recommended pattern for logging is to keep your log messages simple and attach details of the log in parameters. You can attach key/value pairs of strings to your log messages to provide this extra detail, while keeping your logs easy to scan through in a list of messages.
+
+```swift
+log("Said hello to user", parameters: ["user_name": user.name])
+```
+
+```objc
+ARKLogWithParameters(@{ @"user_name": [user name] }, @"Said hello to user");
+```
+
+## Using Dependency Injection
+
+If your app is architected around dependency injection, you can inject an `ARKLogDistributor` instead of using the global logging functions.
+
+```swift
+func sayHello(to name: String, on logDistributor: ARKLogDistributor) {
+    logDistributor.log("Hello \(name)")
+}
+```
+
+```objc
+- (void)sayHelloToPersonNamed:(NSString *)name onLogDistributor:(ARKLogDistributor *)logDistributor;
```

Comment:
extreme nit: the signature of this method is not the same as the signature of the swift method above. They don't have to be. But...
![giphy](https://user-images.githubusercontent.com/139364/108608628-1aa41800-737d-11eb-8ab2-fe5546ce78c5.gif)

Reply:
😂 Fair enough. Updated the Swift signature.